src/cairoUtilFuncGen/memory/arrayLiteral.ts

```diff
@@ -117,44 +113,53 @@ export class MemoryArrayLiteralGen extends StringIndexedFuncGen {
     return funcDef;
   }
 
-  private getOrCreate(type: TypeNode, size: number, dynamic: boolean): GeneratedFunctionInfo {
-    const elementCairoType = CairoType.fromSol(type, this.ast);
+  private getOrCreate(elementType: TypeNode, arraySize: number): GeneratedFunctionInfo {
+    const elementCairoType = CairoType.fromSol(elementType, this.ast);
+    const dynamic = isDynamicArray(elementType) || elementType instanceof StringLiteralType;
+
     const funcName = `wm${this.generatedFunctionsDef.size}_${dynamic ? 'dynamic' : 'static'}_array`;
+    const argString = mapRange(arraySize, (n) => `e${n}: ${elementCairoType.toString()}`).join(
+      ', ',
+    );
 
-    const argString = mapRange(size, (n) => `e${n}: ${elementCairoType.toString()}`).join(', ');
+    const alloc_len: number = dynamic
+      ? arraySize * elementCairoType.width + 1
+      : arraySize * elementCairoType.width;
 
-    // If it's dynamic we need to include the length at the start
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const alloc_len = dynamic ? size * elementCairoType.width + 2 : size * elementCairoType.width;
+    const funcCall =
+      elementCairoType instanceof CairoFelt
+        ? this.requireImport(...WM_WRITE)
+        : this.requireImport(...WM_STORE);
     const writes = [
-      ...(dynamic ? [`wm_write_256{warp_memory=warp_memory}(start, ${uint256(size)});`] : []),
-      ...mapRange(size, (n) => elementCairoType.serialiseMembers(`e${n}`))
+      dynamic ? `warp_memory.write(start, ${arraySize});` : '',
+      ...mapRange(arraySize, (n) => elementCairoType.serialiseMembers(`e${n}`))
         .flat()
         .map(
-          // TODO: use create from WarpMemoryAccessor instead
-          (name, index) => `warp_memory.write(start,
-            ${add('start', dynamic ? index + 2 : index)},
+          (name, index) => `warp_memory.store(
+            ${add('start', index)},
             ${name}
           );`,
         ),
     ];
+    const calledFuncs: FunctionDefinition[] = dynamic
+      ? [this.requireImport(...WM_WRITE), funcCall]
+      : [funcCall];
+
     return {
       name: funcName,
       code: endent`
-        #[implicits(warp_memory: WarpMemory)]
+        #[implicit(warp_memory: WarpMemory)]
```

Comment:
Why? What is the difference?

Reply:
`implicitS` is a reserved word by Cairo. `implict` is the attribute that triggers our plugin. It was misswritten.